packager/packager.cc

```diff
@@ -0,0 +1,665 @@
+                               const UnorderedMap<core::NameRef, shared_ptr<PackageInfo>> &packageInfoByMangledName) {
+    ast::ClassDef::RHS_store importedPackages;
+
+    // Sanity check: Packages _must_ be typed: strict
+    if (file.file.data(ctx).strictLevel != core::StrictLevel::Strict) {
+        if (auto e = ctx.beginError(firstLineOfFile(ctx), core::errors::Packager::PackageFileMustBeStrict)) {
+            e.setHeader("Package files must be typed: strict");
+        }
+    }
+
+    {
+        UnorderedMap<core::NameRef, core::LocOffsets> importedNames;
+        for (auto imported : package.importedPackageNames) {
+            auto it = packageInfoByMangledName.find(imported.mangledName);
+            if (it == packageInfoByMangledName.end()) {
+                if (auto e = ctx.beginError(imported.loc, core::errors::Packager::PackageNotFound)) {
+                    e.setHeader("Cannot find package `{}`", absl::StrJoin(imported.fullName, "::", NameFormatter(ctx)));
+                }
+                continue;
+            }
+
+            if (importedNames.contains(imported.mangledName)) {
+                if (auto e = ctx.beginError(imported.loc, core::errors::Packager::InvalidImportOrExport)) {
+                    e.setHeader("Duplicate package import `{}`",
+                                absl::StrJoin(imported.fullName, "::", NameFormatter(ctx)));
+                    e.addErrorLine(core::Loc(ctx.file, importedNames[imported.mangledName]),
+                                   "Previous package import found here");
+                }
+            } else {
+                importedNames[imported.mangledName] = imported.loc;
+                const auto &found = *it->second;
+
+                ast::ClassDef::RHS_store exportedItemsCopy;
+                for (const auto &exported : found.exportedItems) {
+                    exportedItemsCopy.emplace_back(exported.deepCopy());
+                }
+
+                ENFORCE(!imported.fullName.empty());
+                // Create a module for the imported package that sets up constant references to exported items.
+                // Use proper loc information on the module name so that `import Foo` displays in the results of LSP
+                // Find All References on `Foo`.
+                importedPackages.emplace_back(ast::MK::Module(imported.loc, core::Loc(ctx.file, imported.loc),
+                                                              imported.getFullNameLiteral(imported.loc), {},
+                                                              std::move(exportedItemsCopy)));
+            }
+        }
+    }
+
+    // Include Kernel in the package.
+    importedPackages.emplace_back(ast::MK::Send1(core::LocOffsets::none(), ast::MK::Self(core::LocOffsets::none()),
+                                                 core::Names::include(), name2Expr(core::Names::Constants::Kernel())));
```

Comment:
Why do we need this? Is the runtime support for pay-server going to dynamically `other.include(Kernel)` somehow?

It should be the case that what Sorbet sees statically matches what would happen when the code runs as closely as possible. I think it'd be ideal if all packages manually included Kernel.

Reply:
I added this because pay-server broke catastrophically without it, so I believe yes, we will dynamically `include Kernel`. Is `Kernel` currently explicitly included somewhere in pay-server, or does Ruby implicitly include it in the top-level namespace?

Reply:
Can you link to the errors? Or show me a test case that would have failed without dynamically including `Kernel`?

Reply:
`test/testdata/packager/packages_import_kernel/` -- it calls `Integer` which is defined on Kernel.

Reply:
How could that `def main` method be called?

Reply:
I would have expected code like this to only show up in scripts that are run from the command line. I don't think that scripts will work if they are packaged (either the script won't be wrapped in an anonymous module, and it will be able to access anything, or it will but the `main` method won't be able to be called, is my guess).

I don't think we should add `include Kernel` before we understand why it was needed for this case. How could the `main` method in that test be called?

Reply:
I'll remove it for now; I didn't fully understand the implication of this change, and added it blindly to "fix" pay-server errors. Thanks for pressing me on this.